src/gs-discovery-feed-content-provider.c

```diff
@@ -162,13 +162,13 @@ search_done_cb (GObject *source,
 	guint i;
 	GVariantBuilder builder;
 	guint app_list_length;
+	g_autoptr(GError) error = NULL;
 	g_autoptr(GsAppList) list = NULL;
 
-	list = gs_plugin_loader_job_process_finish (self->plugin_loader, res, NULL);
+	list = gs_plugin_loader_job_process_finish (self->plugin_loader, res, &error);
 	if (list == NULL) {
-		gs_discovery_feed_installable_apps_complete_get_installable_apps (self->skeleton,
-										  search->invocation,
-										  NULL);
+		g_warning ("Error searching for Discovery Feed Apps: %s", error->message);
+		g_dbus_method_invocation_return_gerror (search->invocation, g_steal_pointer (&error));
```

Comment:
If you're passing ownership of the error to the method, you should use `g_dbus_method_invocation_take_error()`, since `_return_gerror()` makes a copy.

Reply:
Makes sense